site/src/pages/api/gh-icon/[data].ts

```diff
@@ -0,0 +1,278 @@
+                ? previousCommand.y2 - previousCommand.y
+                : previousCommand.y2 - prev.y
+              : 0,
+        };
+        addPath(
+          c,
+          c,
+          `M ${prev.x},${prev.y} ${encodeSVGPath({
+            type: SVGPathData.CURVE_TO,
+            relative: false,
+            x: c.x,
+            y: c.y,
+            x1: prev.x - reflectedCp1.x,
+            y1: prev.y - reflectedCp1.y,
+            x2: c.x2,
+            y2: c.y2,
+          })}`
+        );
+        break;
+      }
+      case SVGPathData.QUAD_TO: {
+        assert(prev);
+        addPath(c, c, `M ${prev.x},${prev.y} ${encodeSVGPath(c)}`);
+        break;
+      }
+      case SVGPathData.SMOOTH_QUAD_TO: {
+        assert(prev);
+        const backTrackCP = (
+          index: number,
+          currentPoint: { x: number; y: number }
+        ): { x: number; y: number } => {
+          const previousCommand = commands[index - 1];
+          if (!previousCommand) {
+            return currentPoint;
+          }
+          if (previousCommand.type === SVGPathData.QUAD_TO) {
+            return {
+              x: previousCommand.relative
+                ? currentPoint.x - (previousCommand.x1 - previousCommand.x)
+                : currentPoint.x - (previousCommand.x1 - currentPoint.x),
+              y: previousCommand.relative
+                ? currentPoint.y - (previousCommand.y1 - previousCommand.y)
+                : currentPoint.y - (previousCommand.y1 - currentPoint.y),
+            };
+          }
+          if (previousCommand.type === SVGPathData.SMOOTH_QUAD_TO) {
+            if (!prevCP) {
+              return currentPoint;
+            }
+            return {
+              x: currentPoint.x - (prevCP.x - currentPoint.x),
+              y: currentPoint.y - (prevCP.y - currentPoint.y),
+            };
+          }
+          return currentPoint;
+        };
+        prevCP = backTrackCP(i, prev);
+        addPath(
+          c,
+          c,
+          `M ${prev.x},${prev.y} ${encodeSVGPath({
+            type: SVGPathData.QUAD_TO,
+            relative: false,
+            x: c.x,
+            y: c.y,
+            x1: prevCP.x,
+            y1: prevCP.y,
+          })}`
+        );
+        break;
+      }
+      case SVGPathData.ARC: {
+        assert(prev);
+        addPath(
+          c,
+          c,
+          `M ${prev.x},${prev.y} A ${c.rX} ${c.rY} ${c.xRot} ${c.lArcFlag} ${c.sweepFlag} ${c.x} ${c.y}`
+        );
+        break;
+      }
+      default: {
+        assertNever(c);
+      }
+    }
+  }
+
+  res.setHeader('Content-Type', 'image/svg+xml');
+  res.status(200).end(
+    [
+      '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">',
+
+      // grid
+      '<rect x=".1" y=".1" width="23.8" height="23.8" rx="1" fill="#fff" stroke="#E2E8F0" stroke-width=".1"/>',
```

Comment:
I think it might be a good idea to extract all of these styling attributes to dedicated CSS classes.

Reply:
I do think we should extract it, not sure how well CSS inside SVG is supported.
```ts
const styles = { gridStrokeWidth: 0.1 }
svgson.stringify({ name: 'rect', attributes: { x: styles.gridStrokeWidth, … } })
```

Reply:
This feels like a good solution, what do you think?
```tsx
import { renderToString } from "react-dom/server";

const Grid = ({ fill, stroke, strokeWidth, radius }: { strokeWidth: number, stroke: string, fill?: string, radius?: number }) => (
  <>
    <rect
      width={24 - strokeWidth / 2}
      height={24 - strokeWidth / 2}
      x={strokeWidth / 2}
      y={strokeWidth / 2}
      fill={fill}
      stroke={stroke}
      strokeWidth={strokeWidth}
      rx={radius}
    />
    <path
      stroke={stroke}
      strokeWidth={strokeWidth}
      d={new Array(Math.floor(24 - 1))
        .fill(null)
        .flatMap((_, i) => [
          `M${strokeWidth / 2} ${i + 1}h${24 - strokeWidth}`,
          `M${i + 1} ${strokeWidth / 2}v${24 - strokeWidth}`,
        ])
        .join("")}
    />
  </>
);

console.log(
  renderToString(
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width={24}
      height={24}
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <Grid strokeWidth={.1} stroke="#e2e8f0" fill="#fff" radius={1} />
    </Grid>
  )
);
```

Reply:
`<style>` elements within SVG files are [fully supported](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/style).
Not sure about the code example above, could we not simply use lines instead of rectangles?

Reply:
You are right, styles are supported :+1: 
![](https://user-images.githubusercontent.com/25524993/219381131-d67da173-5f9e-4b06-90c8-c1cec12a5851.svg)